src/Symfony/Component/Cache/Adapter/RedisAdapter.php

```diff
@@ -194,17 +234,49 @@ protected function doSave(array $values, $lifetime)
             return $failed;
         }
         if ($lifetime > 0) {
-            $this->redis->multi(\Redis::PIPELINE);
-            foreach ($serialized as $id => $value) {
-                $this->redis->setEx($id, $lifetime, $value);
-            }
-            if (!$this->redis->exec()) {
-                return false;
+            if ($this->redis instanceof \RedisArray) {
+                $redis = array();
+                foreach ($serialized as $id => $value) {
+                    if (!isset($redis[$h = $this->redis->_target($id)])) {
+                        $redis[$h] = $this->redis->_instance($h);
+                        $redis[$h]->multi(\Redis::PIPELINE);
+                    }
+                    $redis[$h]->setEx($id, $lifetime, $value);
+                }
+                foreach ($redis as $h) {
+                    if (!$h->exec()) {
+                        $failed = false;
+                    }
+                }
+            } else {
+                $this->pipeline(function ($pipe) use ($serialized, $lifetime) {
+                    foreach ($serialized as $id => $value) {
+                        $pipe->setEx($id, $lifetime, $value);
+                    }
+                });
             }
         } elseif (!$this->redis->mSet($serialized)) {
             return false;
         }
 
         return $failed;
     }
+
+    private function pipeline(\Closure $callback)
+    {
+        if ($this->redis instanceof \Predis\Client) {
+            return $this->redis->pipeline($callback);
+        }
+        $pipe = $this->redis instanceof \Redis && $this->redis->multi(\Redis::PIPELINE);
+        try {
+            $callback($this->redis);
+        } catch (\Exception $e) {
+        }
+        if ($pipe) {
+            $this->redis->exec();
+        }
+        if (isset($e)) {
```

Comment:
Instead of using `isset`, you should declare the `$e` variable and check if it's not null here.